justfile

```diff
@@ -0,0 +1,180 @@
+    cd "$RUST_WORKSPACE"; \
+    cargo clippy --all-targets --locked #TODO DENY WARNINGS ON CI
+
+# Checks rust and dart code, fails if there are any issues on CI
+check: rust-check dart-check
+
+# Checks if dart documentation can be build without issues
+dart-check-doc: dart-build
+    cd "$DART_WORKSPACE"; \
+    dart pub global run dartdoc:dartdoc --no-generate-docs --no-quiet
+
+# Checks if rust documentation can be build without issues
+rust-check_doc: _codegen-order-workaround
+    cd "$RUST_WORKSPACE"; \
+    cargo doc --all-features --no-deps --document-private-items --locked
+
+# Builds dart documentation
+dart-doc *args: dart-build
+    cd "$DART_WORKSPACE"; \
+    dart pub global run dartdoc:dartdoc {{args}}
+
+# Builds rust documentation
+rust-doc *args:
+    cd "$RUST_WORKSPACE"; \
+    cargo doc --all-features --document-private-items --locked {{args}}
+
+# Builds rust and dart documentation
+doc: dart-doc rust-doc
+
+# Checks if documentation can be build without issues
+check_doc: dart-check-doc rust-check_doc
+
+_run-cbindgen: _codegen-order-workaround
+    cd "$RUST_WORKSPACE"; \
+    cargo check
+
+_run-ffigen:
+    cd "$DART_WORKSPACE"; \
+    dart run ffigen --config ffigen.yaml
+
+_run-async-bindgen:
+    cd "$RUST_WORKSPACE"; \
+    "{{justfile_directory()}}/$CARGO_INSTALL_ROOT/bin/async-bindgen-gen-dart" \
+        --ffi-class XaynDiscoveryEngineBindingsFfi \
+        --genesis ../"$DART_WORKSPACE"/lib/src/ffi/genesis.ffigen.dart
+
+_run-build-runner:
+    cd "$DART_WORKSPACE"; \
+    dart run build_runner build --delete-conflicting-outputs
+
+# Builds dart (runs all codegen steps)
+dart-build: _run-cbindgen _run-ffigen _run-async-bindgen _run-build-runner
+
+# Builds rust
+rust-build: _codegen-order-workaround
+    cd "$RUST_WORKSPACE"; \
+    cargo build --locked
+
+# Builds dart and rust
+build: rust-build dart-build
+
+# Tests dart (builds all necessary parts)
+dart-test: rust-build dart-build
+    cd "$DART_WORKSPACE"; \
+    dart test
+
+# Tests rust
+rust-test:
+    cd "$RUST_WORKSPACE"; \
+    cargo test --locked
+
+# Tests dart and rust
+test: rust-test dart-test
+
+# Cleans up all generated files
+clean-files:
+    find . \( -name '*.g.dart' \
+        -or -name '*.freezed.dart' \
+        -or -name '*.ffigen.dart' \
+        -or -name '*.ext.dart' \
+    \) -exec rm '{}' \;
+    -rm "$RUST_WORKSPACE"/bindings/include/*
+    -rm "$RUST_WORKSPACE"/bindings/src/async_bindings/*
+
+# Cleans up rusts build cache
+rust-clean-deps:
+    cd "$RUST_WORKSPACE"; \
+    cargo clean
+
+# Cleans up darts build cache
+dart-clean-deps:
+    find "$DART_WORKSPACE" -type d -name .dart_tool -prune -exec rm -r '{}' \;
+
+# Remvoes all local cargo isntalls
+remove-local-cargo-installs:
+    rm -r "$CARGO_INSTALL_ROOT"
+
+# Removes all local dependency artifacts
+clean-deps: rust-clean-deps dart-clean-deps remove-local-cargo-installs
+
+# Removes all local cached dependencies and generated files
+clean-fully: clean-files clean-deps
```

Comment:
Just `clean`?

Reply:
ok, Through that cleans much more then you normally would ever want.

Reply:
Is probably a matter of opinions, When I want to clean I'm usually looking for something like this.
If we realize we want less we can always redirect this to something else.